src/inet/InetInterface.cpp

```diff
@@ -36,9 +36,14 @@
 #include <support/DLLUtil.h>
 
 #if CHIP_SYSTEM_CONFIG_USE_LWIP
+// The lwip headers trigger -Wconversion warnings, so disable those while
+// including those headers.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
```

Comment:
Just an idea, would it be possible to use "-isystem" instead of "-I" for lwip include directories to ignore warnings in the library?

Reply:
Good question.  I would certainly prefer that for lwip and android headers to scattering this pragma stuff about.  @rwalker-apple @mspang where would we put that?

Reply:
Something like this:

```
diff --git a/third_party/lwip/lwip.gni b/third_party/lwip/lwip.gni
index a8fc73ca..b8a5f8e7 100644
--- a/third_party/lwip/lwip.gni
+++ b/third_party/lwip/lwip.gni
@@ -92,10 +92,14 @@ template("lwip_target") {
   }
 
   config("${lwip_target_name}_base_config") {
-    include_dirs = [
+    _include_dirs = [
       "${lwip_root}/repo/lwip/src/include",
       "include",
     ]
+    cflags = []
+    foreach(include_dir, _include_dirs) {
+      cflags += ["-isystem" + rebase_path(include_dir, root_build_dir)]
+    }
 
     # These options may have overlap with lwipopts.h, however this is harmless
     # as long the options are the same and if they are not the same it's a

```

I'm not sure whether it's a good idea. This code is right in our repo so we can fix warnings there.

Reply:
@mspang As in you think we should fix the warnings in lwip and upstream that?

Reply:
Right, so I recalled correctly.  The issue is not in lwip itself.  It's in other headers that lwip includes.  Specifically, `<lwip/sys.h>` includes `src/lwip/freertos/arch/sys_arch.h` which includes NRF5 SDK headers (e.g. FreeRTOS.h) and those SDK headers trigger the warnings (indirectly; the actual warnings are in `nrf_nvic.h` and `app_util.h` in the SDK).

So maybe the pragma bit should move into `src/lwip/freertos/arch/sys_arch.h` or maybe we should treat the SDK header paths with `-isystem`?

Reply:
Fwiw, treating SDK headers with `-isystem` seems to work well.  @Damian-Nordic @mspang thank you!